library/ripemd160.c

```diff
@@ -460,6 +495,12 @@ int mbedtls_ripemd160_self_test( int verbose )
         mbedtls_printf( "\n" );
 
     return( 0 );
+
+fail:
+    if( verbose != 0 )
+        mbedtls_printf( "failed\n" );
+
+    return( ret );
 }
```

Comment:
This self-test does return dedicated error codes instead of `1`; this should be unified with the behaviour of the other self-tests.